### PR TITLE
feat: update dependencies and remove constraint on pydantic

### PR DESCRIPTION
**Pull request by rbiseck3**
### Description
* The `consistent-deps.sh` was fixed to take into account the ingest dependencies, causing some errors to show up. New constriants were added to make that script pass. 
* Update all requirements without constraint on pydantic, allowing the latest version to be pulled in. 
* `pikepdf` is causing a conflict but there's a fix on their `main` branch, just need for the next release to be published. Opened up a question here to see if we can get that out any sooner: [Do releases happen on a schedule?](https://github.com/pikepdf/pikepdf/discussions/574). For now added `lxml<5` to the constraints. 

A couple optimizations: 
* `constraints.in` renamed to `constraints.txt` since the whole point is all dependencies are already pinned and the file never gets compiled
* `constraints.txt` moved to a `requirements/deps` directory as this never gets compiled by `pip-compile`
* Other dependency files updated to reference the new location of `base.in` and `constraints.txt`
* make file updated since it was originally written to avoid the `base.in` and `constraints.in` file

### CODE REVIEW

File: scripts/consistent-deps.sh [hunk 1]
```diff
@@ -16,11 +16,14 @@ function join_by {
...
 
 # NOTE(alan): Add any dependency files here we don't want to include in the resolution.
-excludefiles=("requirements/build.txt")
+excludefiles=("requirements//build.txt")
 
 # Build an array of requirements files.
 shopt -s nullglob
-reqfiles=(requirements/*.txt)
+reqfiles=()
+while IFS= read -r -d $'\0'; do
```
Comment (MthwRobinson) [inline scripts/consistent-deps.sh:24]: What does this change to the script do?

Comment (MthwRobinson) [reply scripts/consistent-deps.sh:24]: Is this to get the ingest requirements from the subdirectory?

**Review by MthwRobinson — APPROVED** [review]
Updates look good, just had a question on the `consistent-deps.sh` update

File: scripts/consistent-deps.sh (hunk 1 above)
Comment (rbiseck3) [reply scripts/consistent-deps.sh:24]: This creates a bash array out of all the `*.txt` files in here. This was previously missing the ones in the ingest folder.

Comment (MthwRobinson) [reply scripts/consistent-deps.sh:24]: Thanks!

File: requirements/base.in [hunk 2]
```diff
@@ -16,5 +16,5 @@ rapidfuzz
 backoff
 typing-extensions
 # NOTE(jennings): pinned due to later versions not supporting api_key_auth in UnstructuredClient
-unstructured-client>=0.15.1
```
Comment (Coniferish) [inline requirements/base.in:19]: @rbiseck3, do you happen to remember why this version was restricted? The newer parameters of the client aren't accessible with this restricted

**Pull request merged**